### PR TITLE
Feature/onboarding setup

### LINKED ISSUE #8131: [Onboarding] Get started screens
image

**Recovery is not in the scope of this PR**

Acceptance criteria:

1. pixel perfect
1. it's possible to swipe pictures
1. pictures is autoswiped?

Figma: https://www.figma.com/file/dEIljL7UPbXgsZUA0Q4qlE5E/Onboarding?node-id=927%3A14702

### LINKED ISSUE #8132: [Onboarding] Generate keys
image

image

**status-go issue** https://github.com/status-im/status-go/issues/1456

Acceptance criteria:

1. pixel perfect
1. it's possible to generate accounts
1. choose an account

Figma: https://www.figma.com/file/dEIljL7UPbXgsZUA0Q4qlE5E/Onboarding?node-id=927%3A14702

### LINKED ISSUE #8134: [Onboarding] Select storage and save with password
image

**Keycard is not in the scope of this PR**

Acceptance criteria:

1. pixel perfect
1. it's possible to select device storage
1. store with password

Figma: https://www.figma.com/file/dEIljL7UPbXgsZUA0Q4qlE5E/Onboarding?node-id=927%3A14702

### PR DESCRIPTION
**Pull request by vitvly**
Fixes #8131 
Fixes #8132
Fixes #8134

### CODE REVIEW

File: src/status_im/init/core.cljs
Comment (yenda) [inline src/status_im/init/core.cljs]: leftover?

File: src/status_im/subs.cljs
Comment (yenda) [inline src/status_im/subs.cljs]: top level subs should be put at the top of the file

File: src/status_im/ui/components/numpad/views.cljs
Comment (yenda) [inline src/status_im/ui/components/numpad/views.cljs]: is it the same namepad again?

File: src/status_im/ui/screens/intro/styles.cljs
Comment (yenda) [inline src/status_im/ui/screens/intro/styles.cljs]: leftovers?

**Review by yenda — COMMENTED** [review]
i relealized that it was a draft after reviewing

**Comment by flexsurfer** [comment]
@andmironov https://youtu.be/4bZxGZlDc0k

**Comment by flexsurfer** [comment]
@siphiuel could you rebase please onto develop

File: src/status_im/events.cljs
Comment (flexsurfer) [inline src/status_im/events.cljs]: there is no need to register handlers , you can use {:events []} in fx/defn , please move all these handlers in core , thx

File: src/status_im/subs.cljs
Comment (vitvly) [reply src/status_im/subs.cljs]: Done

File: src/status_im/events.cljs
Comment (vitvly) [reply src/status_im/events.cljs]: Done

File: src/status_im/accounts/create/core.cljs
Comment (flexsurfer) [inline src/status_im/accounts/create/core.cljs]: leftover?

File: src/status_im/accounts/create/core.cljs
Comment (flexsurfer) [inline src/status_im/accounts/create/core.cljs]: leftover?

File: src/status_im/accounts/create/core.cljs
Comment (flexsurfer) [inline src/status_im/accounts/create/core.cljs]: better to use update :intro-wizard and assoc

File: src/status_im/accounts/create/core.cljs
Comment (flexsurfer) [inline src/status_im/accounts/create/core.cljs]: you can move all keys under one assoc

File: src/status_im/ui/screens/about_app/styles.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/about_app/styles.cljs]: you can use typography instead

File: src/status_im/ui/screens/about_app/styles.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/about_app/styles.cljs]: this is default typography no need to specify font size and line height

File: src/status_im/ui/screens/intro/views.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/intro/views.cljs]: leftover?

File: src/status_im/ui/screens/intro/views.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/intro/views.cljs]: leftover?

File: src/status_im/ui/screens/intro/views.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/intro/views.cljs]: leftover?

File: src/status_im/ui/screens/profile/tribute_to_talk/views.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/profile/tribute_to_talk/views.cljs]: how's this related to onboarding?

File: src/status_im/ui/screens/profile/tribute_to_talk/styles.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/profile/tribute_to_talk/styles.cljs]: how's this related to onboarding?

File: src/status_im/utils/utils.cljs
Comment (flexsurfer) [inline src/status_im/utils/utils.cljs:22]: how's this related to onboarding?

**Review by flexsurfer — CHANGES_REQUESTED** [review]
1) clean log
2) use typography
3) remove code not related to onboarding (ttt)

**Comment by vitvly** [comment]
Welcome screen:
Simulator Screen Shot - iPhone Xs - 2019-06-13 at 12 16 34

Generate key:
Simulator Screen Shot - iPhone Xs - 2019-06-13 at 12 16 36

Choose key:
Simulator Screen Shot - iPhone Xs - 2019-06-13 at 12 16 38

Choose storage:
Simulator Screen Shot - iPhone Xs - 2019-06-13 at 12 16 41

Create password:
Simulator Screen Shot - iPhone Xs - 2019-06-13 at 12 31 46

Confirm password:
Simulator Screen Shot - iPhone Xs - 2019-06-13 at 12 31 56

Confirm password error:
Simulator Screen Shot - iPhone Xs - 2019-06-13 at 12 31 37

Enable fingerprint:
Simulator Screen Shot - iPhone Xs - 2019-06-13 at 12 17 07

Enable notifications:
Simulator Screen Shot - iPhone Xs - 2019-06-13 at 12 17 10

**Comment by vitvly** [comment]
Small screen:

Welcome:
Simulator Screen Shot - iPhone SE - 2019-06-13 at 13 22 03

Generate key:
Simulator Screen Shot - iPhone SE - 2019-06-13 at 13 22 08

Select account:
Simulator Screen Shot - iPhone SE - 2019-06-13 at 13 22 10

Choose key storage:
Simulator Screen Shot - iPhone SE - 2019-06-13 at 13 22 12

Create password:
Simulator Screen Shot - iPhone SE - 2019-06-13 at 13 22 16

Confirm password:
Simulator Screen Shot - iPhone SE - 2019-06-13 at 13 22 20

Confirm error:
Simulator Screen Shot - iPhone SE - 2019-06-13 at 13 22 24

Fingerprint:
Simulator Screen Shot - iPhone SE - 2019-06-13 at 13 22 35

Notifications:
Simulator Screen Shot - iPhone SE - 2019-06-13 at 13 22 37

File: src/status_im/init/core.cljs
Comment (vitvly) [reply src/status_im/init/core.cljs]: Removed.

File: src/status_im/ui/components/numpad/views.cljs
Comment (vitvly) [reply src/status_im/ui/components/numpad/views.cljs]: Yes, but this will be now in a separate PR, together with PINs logic. Removed from this PR now.

File: src/status_im/ui/screens/intro/styles.cljs
Comment (vitvly) [reply src/status_im/ui/screens/intro/styles.cljs]: Removed.

File: src/status_im/accounts/create/core.cljs
Comment (vitvly) [reply src/status_im/accounts/create/core.cljs]: Removed.

File: src/status_im/accounts/create/core.cljs
Comment (vitvly) [reply src/status_im/accounts/create/core.cljs]: Removed.

File: src/status_im/accounts/create/core.cljs
Comment (vitvly) [reply src/status_im/accounts/create/core.cljs]: Thanks, yes looks better, fixed.

File: src/status_im/accounts/create/core.cljs
Comment (vitvly) [reply src/status_im/accounts/create/core.cljs]: Sorry, so lame, fixed.

File: src/status_im/ui/screens/profile/tribute_to_talk/views.cljs
Comment (vitvly) [reply src/status_im/ui/screens/profile/tribute_to_talk/views.cljs]: Leftover from numpad migration to separate PR, removed.

File: src/status_im/utils/utils.cljs
Comment (vitvly) [reply src/status_im/utils/utils.cljs:22]: `Passcodes don't match` error message on `Confirm code` screen should trigger vibration as per Figma design.

File: src/status_im/ui/screens/intro/views.cljs
Comment (vitvly) [reply src/status_im/ui/screens/intro/views.cljs]: Removed.

File: src/status_im/ui/screens/intro/views.cljs
Comment (vitvly) [reply src/status_im/ui/screens/intro/views.cljs]: Removed.

File: src/status_im/ui/screens/profile/tribute_to_talk/styles.cljs
Comment (vitvly) [reply src/status_im/ui/screens/profile/tribute_to_talk/styles.cljs]: In no way indeed, my mistake, removed.

File: src/status_im/ui/screens/about_app/styles.cljs
Comment (vitvly) [reply src/status_im/ui/screens/about_app/styles.cljs]: Done.

File: src/status_im/ui/screens/about_app/styles.cljs
Comment (vitvly) [reply src/status_im/ui/screens/about_app/styles.cljs]: Do you mean i can remove `line-height` and rely on automatic calculation, or something else?

Comment (flexsurfer) [reply src/status_im/ui/screens/about_app/styles.cljs]: i mean we don't use styles for text anymore, only `:typography` because all labels are standatized so instead of `[react/text {:style}]` you should use `[react/text {:typography}`

Comment (vitvly) [reply src/status_im/ui/screens/about_app/styles.cljs]: Thanks for clarifying, fixed.

File: src/status_im/ui/screens/intro/views.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/intro/views.cljs]: view can be removed?

Comment (vitvly) [reply src/status_im/ui/screens/intro/views.cljs]: Done.

File: src/status_im/ui/screens/profile/tribute_to_talk/views.cljs
Comment (flexsurfer) [reply src/status_im/ui/screens/profile/tribute_to_talk/views.cljs]: still here

Comment (vitvly) [reply src/status_im/ui/screens/profile/tribute_to_talk/views.cljs]: Removed for good :)

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/accounts/create/core.cljs
Comment (jeluard) [inline src/status_im/accounts/create/core.cljs]: Out of scope but keywords would be more visible

File: src/status_im/accounts/create/core.cljs
Comment (jeluard) [inline src/status_im/accounts/create/core.cljs]: This whole fn would benefit from some smaller functions

File: src/status_im/native_module/core.cljs
Comment (jeluard) [inline src/status_im/native_module/core.cljs]: Why log?

File: src/status_im/native_module/impl/module.cljs
Comment (jeluard) [inline src/status_im/native_module/impl/module.cljs]: Same

File: src/status_im/ui/screens/intro/styles.cljs
Comment (jeluard) [inline src/status_im/ui/screens/intro/styles.cljs]: Magic?

**Review by jeluard — APPROVED** [review]

File: src/status_im/accounts/create/core.cljs
Comment (vitvly) [reply src/status_im/accounts/create/core.cljs]: Agree, converted to keywords

File: src/status_im/accounts/create/core.cljs
Comment (vitvly) [reply src/status_im/accounts/create/core.cljs]: Done.

File: src/status_im/native_module/core.cljs
Comment (vitvly) [reply src/status_im/native_module/core.cljs]: Removed.

File: src/status_im/native_module/impl/module.cljs
Comment (vitvly) [reply src/status_im/native_module/impl/module.cljs]: Removed.

File: src/status_im/ui/screens/intro/styles.cljs
Comment (vitvly) [reply src/status_im/ui/screens/intro/styles.cljs]: Added surrounding let giving meaningful names to numbers 6 and 10.

**Comment by hesterbruikman** [comment]
@siphiuel not sure if this is the best time, please let me know as @errorists is out today. I notice the error message on the Password entry screen says "Passcodes don't match" For the Password screen this should be "Passwords don't match" For the Passcode screen this can remain as it.

**Review by jeluard — APPROVED** [review]

**Review by jakubgs — APPROVED** [review]
I see no `mobile_files/yarn.lock` changes so this bot is either retarded, or they were reversed, either way this should shut it up.

**Comment by vitvly** [comment]
@hesterbruikman Corrected password error label

**Comment by rachelhamlin** [comment]
@siphiuel are there any blockers remaining here? Can we merge?

**Comment by errorists** [comment]
I'm doing some testing atm, some issues cropped up with the scrollable image carousel that I didn't notice when on the video call with Vitaliy, sorry! just to recap them here:

- There's noticeable padding on both sides of the image carousel when scrolling. The carousel should stretch the entire width of the screen.
- The dots below the carousel should be switched on/off in the same event - when page has changed. Currently it looks like when on scroll start they switch off and if the page changed, they switch back on.
- there's also something weird happening when I try to scroll right on the last page and to the left on the first, if it's because we want the carousel to be 'infinite', i'd disable that. here's a video https://www.dropbox.com/s/gntlyrv6w4gjy3q/vitaliy.mp4?dl=0

**Comment by yenda** [comment]
@errorists do you think we could merge and fix these issues separately?

**Comment by errorists** [comment]
@yenda sure 👍

**Comment by vitvly** [comment]
This last one is me failing at magic tricks, just removed it.

**Comment by vitvly** [comment]
Pushed an updated branch with status-go version set to 0.29.0-beta.1, as https://github.com/status-im/status-go/pull/1464 is merged now.

**Comment by churik** [comment]
> @siphiuel are there any blockers remaining here? Can we merge?

@rachelhamlin it is not tested yet by QA team, I believe it should.

**Comment by asemiankevich** [comment]
@siphiuel all builds are failed, please take a look %(

File: src/status_im/ui/screens/routing/screens.cljs
Comment (jakubgs) [inline src/status_im/ui/screens/routing/screens.cljs]: An importing error?
```
Compiling ["index.android.js" "status-modules/cljs/i18n-raw.js" "status-modules/cljs/network-raw.js" "status-modules/cljs/extensions-raw.js"] from ["components/src" "react-native/src/cljsjs" "react-native/src/mobile" "src" "env/prod" "prod"]...
WARNING: Use of undeclared Var status-im.ui.screens.intro.views/intro-wizard at line 74 /home/jenkins/workspace/status-react_prs_android_PR-8234/src/status_im/ui/screens/routing/screens.cljs
Error encountered performing task 'cljsbuild' with profile(s): 'prod'
```

Comment (vitvly) [reply src/status_im/ui/screens/routing/screens.cljs]: Fixed, post-rebase issue.

**Comment by vitvly** [comment]
@asemiankevich rebased with latest develop and made fixes needed after migration to status-go 0.29.0-beta.1 library

**Comment by vitvly** [comment]
Sorry, see that haven't properly removed enable-installation calls from status-react that were removed in status-go. will fix

File: translations/en.json
Comment (yenda) [inline translations/en.json]: typo `enambling`

**Comment by vitvly** [comment]
@asemiankevich Builds fixed now, apologies for long delay

**Comment by asemiankevich** [comment]
@Serhy ^ :"(

**Comment by Serhy** [comment]
On both iOS (12.3) and Android (8.0, 8.1) I get the next error when I tap `Generate a key` button:
Confirmed with the latest builds: https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190703-163732-46a4e8-pr.apk (iOS Diawii - https://i.diawi.com/Jq7dgj)
Screenshot 2019-07-04 at 10 58 13

1) Install fresh Status build
2) Open Status
3) Tap `Get Started`
4) Tap `Generate a key` button

**Comment by flexsurfer** [comment]
@siphiuel could you take a look?

**Comment by asemiankevich** [comment]
@siphiuel also when recovering account -> i stay on recovery screen with empty fields
image

**Comment by vitvly** [comment]
@Serhy @asemiankevich fixed the "Generating keys..." and recovery screen issues

**Comment by asemiankevich** [comment]
@siphiuel can't set up new account, error shown when confirming password
image

also, password input screen is cut as well as terms and conditions on the main screen
image

image

**Comment by vitvly** [comment]
Fixed confirm screen error and bottom margin on main screen.

**Comment by asemiankevich** [comment]
@siphiuel it looks a bit different from mocks (icons should be bright and colorful)

image

Also [Secure] should be with capital letter

image

I think the icons are now very close to the top of the screen because you implemented bottom margin?

@errorists @hesterbruikman your review please 🙏

**Comment by asemiankevich** [comment]
@siphiuel i still can repro this error, when i confirm password with empty value

https://user-images.githubusercontent.com/28752642/60663624-692f3680-9e68-11e9-9392-885a39af66d5.png

also i get this one right after ^

image

**Comment by asemiankevich** [comment]
also, i dont see those screens at all after i create account @siphiuel 
image

**Comment by vitvly** [comment]
We had a discussion earlier with @errorists that the last two screens can be omitted for now.

**Comment by errorists** [comment]
Correct. Welcome screen is dependant on the illustration that we're working on. It will be some text and a button, without the spinner. 

The updated chat empty state[ can be found here](https://www.figma.com/file/dEIljL7UPbXgsZUA0Q4qlE5E/Onboarding?node-id=1669%3A42) and I'd suggest to leave it to a separate PR

**Comment by vitvly** [comment]
Regarding capitalization of titles on the main screen - should we strictly follow Figma design (words capitalized on the first slide only), or capitalize all slide titles instead?

**Comment by asemiankevich** [comment]
@siphiuel i am sorry , this long thread is hard to read. Will you be able to fix the error on password?
I still get it in last builds

image

**Comment by vitvly** [comment]
@asemiankevich i've temporarily commented out the vibrate fn reference. Not sure how much time it'll take to fix, so maybe we can fix it in a separate PR

**Comment by flexsurfer** [comment]
separate PR sounds good

**Comment by asemiankevich** [comment]
@siphiuel 

1. `null is not an object` error when removing password on android (android 8 in my case).
image

2. when i enter 6 digits on android -> i can't proceed, Next button is not active
image

Please, take a look.

**Comment by vitvly** [comment]
Fixes create-password screen issues.

**Comment by hesterbruikman** [comment]
Onboarding screens look great everyone!

@siphiuel  I noticed that the Notifications don't trigger native notification settings yet. Is this out of this PR? Captured anywhere else?

**Comment by asemiankevich** [comment]
> Onboarding screens look great everyone!
> 
> @siphiuel I noticed that the Notifications don't trigger native notification settings yet. Is this out of this PR? Captured anywhere else?

it should be out of scope of this PR, we may open separate GHI for this @hesterbruikman  @flexsurfer

**Comment by asemiankevich** [comment]
@siphiuel would you mind to create a separate task for Fingerprint and notification pieces if they are not in scope of this PR? https://github.com/status-im/status-react/pull/8234#issuecomment-508795992

**Comment by vitvly** [comment]
@asemiankevich  created an issue https://github.com/status-im/status-react/issues/8533 and referenced it in master issue https://github.com/status-im/status-react/issues/8089

**Comment by vitvly** [comment]
Fixed password input text cutting.

**Comment by flexsurfer** [comment]
@siphiuel could you rebase and squash commits please

**Comment by vitvly** [comment]
@flexsurfer done

**Comment by asemiankevich** [comment]
@Serhy awesome job ! Thank you!

**Comment by Serhy** [comment]
@siphiuel please mind latest commits fixing autotests before merge

**Comment by hesterbruikman** [comment]
@rachelhamlin @flexsurfer 

Noticed some UI issues in the onboarding setup. As this PR is merged, where would you prefer I report these? One seperate issue, multiple seperate issues, or add to an existing issue?:
- Notification icon is cut off at the bottom
- Navigation is too high, should be in header 44px from the top
- Keycard icon scales on Samsung Galaxy x and becomes unreadible
- Addresses on account selection need to be truncated further (0x8ueq...8ujj)

Alternatively this can also wait untill we have the illustrations and update in one go. Just want to make sure these don't get lost.

Some issues already reported earlier by @errorists but outside of PR scope:
- There's noticeable padding on both sides of the image carousel when scrolling. The carousel should stretch the entire width of the screen.
- The dots below the carousel should be switched on/off in the same event - when page has changed. Currently it looks like when on scroll start they switch off and if the page changed, they switch back on.

cc @andmironov @errorists

**Comment by flexsurfer** [comment]
filed new issue https://github.com/status-im/status-react/issues/8551

**Comment by andmironov** [comment]
Hey, the UI/UX issues I've spotted:

**Welcome screens**
- The welcome screen cuts off the images and text, should not be this way (see screenshot below);
- on "Get yourself a key" screen, when tapping on "Generate a key" button, the image moves a few pixels down;
- If I tap "Generate a key" -> See the list of available account names -> tap "back" -> tap "Generate a key" again -> there's a bug: infinite spinner "Generating a key";

**Choose a key and name screen**
- Wrong layout (margins around screen title and the list) (see Figma);
- Bottom sheet (the one you see after tapping on "Learn more") has wrong margins  (see Figma);
- Wrong margins In list items (see Figma);
- The address in list items is too long (should be 8 characters, see Figma);
- The line that separates the "Next" button is not 100% wide  (see Figma);

**Select key storage screen**
- Slightly wrong layout (see Figma);
- List items have slightly wrong margins  (see Figma);
- Bottom sheet (the one you see after tapping on "Learn more") has wrong margins  (see Figma);


**Create password screen**
- The cursor follows the opening keyboard, overlays the text;
- The line that separates the "Next" button is not 100% wide  (see Figma);
- The typeface of the password is too big;
- The margin around the "Next" button is too big (see Figma);

**Enable fingerprint**
- I have an iPhone X, why am I seeing this?;
- The "Sync using mobile data" bottom sheet appears here. Does not make sense, I think we should show it after the setup is complete, maybe after the second launch of the app;

**Enable notifications**
- I tap "Enable" and do not see the native notification's dialog;

photo_2019-07-10 14 52 03

**Comment by hesterbruikman** [comment]
> 
> 
> Hey, the UI/UX issues I've spotted:

Copied here: https://github.com/status-im/status-react/issues/8551
Notification fix is a PR in review #8547

**Pull request merged**
